### PR TITLE
ui: add document importation through the web

### PR DESCRIPTION
**Pull request by jma**
Allows the librarian to search for an existing record on a remote
catalog (such as the BnF catalog, through the SRU endpoint), to identify
the desired record, to load the data in the editor in order to modify it
and finally to add it to the RERO ILS catalog.

* Removes the existing import function in the editor.
* Adds BNF importation.
* Adds URL parameters to import an external source in the editor.
* Adds an import button in the Import document detailed view.
* Adds the total number of results found on the remote server.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on `rero-ils`'s PR(s):

* rero/rero-ils#1059
* rero/ng-core#194

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.ts [hunk 1]
```diff
@@ -46,16 +51,43 @@ export class DocumentDetailViewComponent implements DetailRecord, OnInit, OnDest
    * @param _translateService: TranslateService
    */
   constructor(
-    private _translateService: TranslateService
+    private _translateService: TranslateService,
+    private _router: ActivatedRoute,
+    private _recordService: RecordService
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.ts:56]: COSMETIC : Missing docString

File: projects/admin/src/app/routes/import-documents-route.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  /** Route name */
+  readonly name = 'import';
+
+  /**
+   * Constructor
+   * @param routeToolService - RouteToolService
+   */
+  constructor(
+    protected _routeToolService: RouteToolService,
+    private _translateService: TranslateService
```
Comment (zannkukai) [inline projects/admin/src/app/routes/import-documents-route.ts:37]: COSMETIC : Missing params docString

File: projects/admin/src/app/routes/import-documents-route.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    };
+  }
+  /**
+   *
+   * @param hits list of hit results.
+   * @return observable of the string representation of the number of results.
+   */
+  getResultsText(hits): Observable<string> {
+    if (hits.total === 0) {
+      return this._translateService.stream('no result');
```
Comment (zannkukai) [inline projects/admin/src/app/routes/import-documents-route.ts:89]: What is the difference between 'stream' and 'instant' ?

File: projects/admin/src/app/routes/route.service.ts [hunk 4]
```diff
@@ -50,7 +52,8 @@ export class RouteService {
   constructor(
     private _routeCollectionService: RouteCollectionService,
     private _router: Router,
-    private _routeToolService: RouteToolService
+    private _routeToolService: RouteToolService,
+    private _translateService: TranslateService
```
Comment (zannkukai) [inline projects/admin/src/app/routes/route.service.ts:56]: missing docstring for this param

File: projects/admin/src/app/routes/import-documents-route.ts (hunk 3 above)
Comment (jma) [reply projects/admin/src/app/routes/import-documents-route.ts:89]: The former returns a observable and the later the string.

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html [hunk 5]
```diff
@@ -201,43 +218,35 @@ <h3>{{ altgr_title }}</h3>
...
-                </dt>
-                <dd [ngClass]="ddCssClass">
-                  <ul class="list-unstyled mb-0">
-                    <li
-                      *ngFor="let note of notes.accompanyingMaterial"
-                    >{{ note }}</li>
-                  </ul>
-                </dd>
-              </ng-container>
+            <ng-container *ngIf="record.metadata.notes_text && record.metadata.notes_text.general">
```
Comment (AoNoOokami) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html:221]: `notes_text` doesn't exist any more after the quick fix I merged recently. The correct way to display the notes is the pipe `formatNotes`.

**Comment by jma** [comment]
> ui: add document importation through the web 
> 
> Allows the librarian to search for an existing record on a remote
> catalog (such as the BnF catalog, through the SRU endpoint), to identify
> the desired record, to load the data in the editor in order to modify it
> and finally to add it to the RERO ILS catalog.
> 
> * Removes the existing import function in the editor.
> * Adds BNF importation.
> * Adds URL parameters to import an external source in the editor.
> * Adds an import button in the Import document detailed view.
> * Adds the total number of results found on the remote server.

Done thank you.

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html (hunk 5 above)
Comment (jma) [reply projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html]: I made a mistake with a conflict resolution during a rebase. I hope that it is the only one...

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html [hunk 6]
```diff
@@ -112,16 +111,36 @@ <h3>{{ altgr_title }}</h3>
...
+                  <td>{{ f.ind1 }}</td>
+                  <td>{{ f.ind2 }}</td>
+                  <td>{{ f.value }}</td>
+                </ng-container>
+              </tr>
+            </tbody>
+          </table>
         </div>
       </tab>
       <!-- END OF GET TAB ----------------------------------------------------->
```
Comment (AoNoOokami) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html:146]: End of marc tab?

**Pull request merged**